### PR TITLE
Add host, seed, & teardown flags with asc. implementation

### PR DESCRIPTION
**Pull request by desa**

### CODE REVIEW

File: etc/mock_data_dir/cat1/fuction_select/pretty_count_star.json [hunk 1]
```diff
@@ -0,0 +1,21 @@
...
+            [
+              0,
+              15258
+            ]
+          ]
+        }
+      ]
+    }
+  ]
+}
```
Comment (mark-rushakoff) [inline etc/mock_data_dir/cat1/fuction_select/pretty_count_star.json:21]: What's the purpose of `pretty_count_star`? If it's just to indicate that the .json files don't care about whitespace, then I think it ought to be different from an existing query/result.

File: data_tests/queries [hunk 2]
```diff
@@ -0,0 +1,265 @@
+SELECT * FROM "h2o_feet" LIMIT 3'
```
Comment (mark-rushakoff) [inline data_tests/queries]: Is this whole file the queries from the docs? This file would currently be ignored by the tool, right?

File: data/category.go [hunk 3]
```diff
@@ -169,7 +183,7 @@ func (s *specification) Test(cfg write.ClientConfig) error {
 	}
 
 	if !eq {
-		return fmt.Errorf("expected:\n%s\vgot:\n%s\n", exp, got)
+		return fmt.Errorf("Query: %v\nExpected: \n%s\vGot: \n%s\n", string(q), exp, got)
```
Comment (mark-rushakoff) [inline data/category.go]: This seems fine for now, but eventually we should have something that produces a more intelligent, JSON-aware diff. I don't know of any such library off the top of my head.

File: data/category.go [hunk 4]
```diff
@@ -116,17 +119,24 @@ func (c *Category) Seed(cfg write.ClientConfig) (int, error) {
 	}
 }
 
-func (c *Category) Test(cfg write.ClientConfig) error {
+func (c *Category) Test(cfg write.ClientConfig) (err error) {
```
Comment (mark-rushakoff) [inline data/category.go]: This API is hard to follow, and I think it's in part due to lack of GoDoc.

This method assigns to the err variable but always returns nil.  Is the error supposed to indicate that something went legitimately wrong, or is supposed to indicate a test failure?

File: data/category.go [hunk 5]
```diff
@@ -18,7 +18,7 @@ import (
 	"github.com/influxdata/influx-stress/write"
 )
 
-func GetDataDirs(root, pattern string) []string {
+func GetDataDirs(root string) []string {
```
Comment (mark-rushakoff) [inline data/category.go]: I don't like the name Data here because that word doesn't really inform the reader of anything specific.

Can we rename Data to Select? I think all the tests under here are ultimately select statements, right?

File: data/category.go [hunk 6]
```diff
@@ -31,10 +31,13 @@ func GetDataDirs(root, pattern string) []string {
 	return specs
 }
 
-func NewCategories(dirs []string) []spec.Spec {
+func NewCategories(dirs []string, filter string) []spec.Spec {
```
Comment (mark-rushakoff) [inline data/category.go]: I'm not sure about the Categories name either. Right now a Category contains a dataset and a collection of queries and results.

A `Suite` would be slightly more common language in testing, but I don't think that's quite exact either. 

Category is probably fine for now, but think about what might be a better name.

File: data/category.go [hunk 7]
```diff
@@ -116,17 +119,24 @@ func (c *Category) Seed(cfg write.ClientConfig) (int, error) {
...
 }
 
-func (c *Category) Test(cfg write.ClientConfig) error {
+func (c *Category) Test(cfg write.ClientConfig) (err error) {
 	for _, s := range c.specs {
-		err := s.Test(cfg)
+		err = s.Test(cfg)
 		if err != nil {
-			fmt.Println(err)
+			fmt.Fprint(os.Stderr, err)
```
Comment (mark-rushakoff) [inline data/category.go]: I think rather than writing directly to stderr, a `Category` should have an associated SuccessRecorder and FailureRecorder. `(spec).Test` should not be aware of result recorders, but should just return its result.

Something like

``` go
res, err := s.Test(cfg)
if err != nil {
  fmt.Fprintf(os.Stderr, "Error executing test: %v\n", err)
}
if res.Success {
 c.RecordSuccess(res.Name)
} else {
 c.RecordFailure(res.Failure)
}
```

Maybe not that API precisely, but something along those lines.  That will make it easier going forward to adjust output accordingly, and could potentially support jUnit XML, TAP, or whatever format other tooling needs to consume.

File: data/category.go [hunk 8]
```diff
@@ -140,6 +150,10 @@ func (s *specification) Seed(cfg write.ClientConfig) error {
 	return nil
 }
 
+func (s *specification) Teardown(cfg write.ClientConfig) error {
```
Comment (mark-rushakoff) [inline data/category.go:168]: Can an individual spec ever have a teardown? Method might not be necessary.

**Review by mark-rushakoff — CHANGES_REQUESTED** [review]
Added a couple questions and thoughts. More godoc in general will be helpful here too.

File: etc/mock_data_dir/cat1/fuction_select/pretty_count_star.json (hunk 1 above)
Comment (desa) [reply etc/mock_data_dir/cat1/fuction_select/pretty_count_star.json:21]: everything in the `/mock_data_dir` directory was just meant to be for testing that walking the directory structure works correctly.

File: data_tests/queries (hunk 2 above)
Comment (desa) [reply data_tests/queries]: Whoops. Accidental commit. I'm just was keeping a log of all of the queries from the docs as a transitioned them into `.query` and `.json` files.

File: data/category.go (hunk 5 above)
Comment (desa) [reply data/category.go]: There's a couple meta queries that are also a part of this. `show series`, `show field keys`, `show tag keys`, and all of the variants of those kinds of queries. 

I agree that data sounds wrong though. We could add the restriction that they be for selects and have a separate command for meta queries.

File: data/category.go (hunk 6 above)
Comment (desa) [reply data/category.go]: I like `Suite`.

File: data/category.go (hunk 4 above)
Comment (desa) [reply data/category.go]: I was imagining that `err` was when something legitimately went wrong. I'll add some GoDoc to clarify though.

File: data/category.go (hunk 7 above)
Comment (desa) [reply data/category.go]: Yup

File: data/category.go (hunk 8 above)
Comment (desa) [reply data/category.go:168]: Had it just so that specification implemented the `Spec` interface, probably doesn't need to though.

File: data/category.go (hunk 3 above)
Comment (desa) [reply data/category.go]: Agreed

**Comment by desa** [comment]
@mark-rushakoff fixed most of the issues you brought up (at least I think I did). There's still more work that needs to be done, but I think it's outside the scope of this PR.

**Review by mark-rushakoff — APPROVED** [review]
